packages/xo-web/src/common/select-objects.js

```diff
@@ -250,23 +250,32 @@ class GenericSelect extends React.Component {
   }
 
   // GroupBy: Display option with margin if not disabled and containers exists.
-  _renderOption = option => (
-    <span
-      className={
-        !option.disabled && this.props.xoContainers !== undefined
-          ? 'ml-1'
-          : undefined
-      }
-    >
-      {renderXoItem(option.xoItem, {
-        type:
-          this.props.resourceSet !== undefined &&
-          option.xoItem.type !== undefined
-            ? `${option.xoItem.type}-resourceSet`
-            : undefined,
-      })}
-    </span>
-  )
+  _renderOption = option => {
+    const { xoItem } = option
+    const { id, type } = xoItem
+    const _type =
+      this.props.resourceSet !== undefined && option.xoItem.type !== undefined
+        ? `${option.xoItem.type}-resourceSet`
+        : undefined || type
+    const Component = xoItemToRender[_type]
+
+    return (
+      <span
+        className={
+          !option.disabled && this.props.xoContainers !== undefined
+            ? 'ml-1'
+            : undefined
+        }
+        key={id}
+      >
+        {type === 'host' ? (
+          <Host id={id} memoryFree />
+        ) : (
+          <Component {...xoItem} />
+        )}
```

Comment:
I think it will be too much of a mess to add something on another component later. However, we could pass custom props to [`renderXoItem`](https://github.com/Rajaa-BARHTAOUI/xo-web/blob/ram_left_on_migrateVm/packages/xo-web/src/common/render-xo-item.js#L495) like `(item, { className, type: xoType, ...props } = {}) =>` and then pass them to `<Component {...props} />`. What do you think?